packages/xrpl/src/Wallet/index.ts

```diff
@@ -377,6 +382,29 @@ class Wallet {
     return classicAddressToXAddress(this.classicAddress, tag, isTestnet)
   }
 
+  /**
+   * Remove trailing insignificant zeros for non-XRP Payment amount.
+   * This resolves the serialization mismatch bug when encoding/decoding a non-XRP Payment transaction
+   * with an amount that contains trailing insignificant zeros; for example, '123.4000' would serialize
+   * to '123.4' and cause a mismatch.
+   *
+   * @param tx - The transaction prior to signing.
+   */
+  // eslint-disable-next-line class-methods-use-this -- Helper for organization purposes
+  private removeTrailingZeros(tx: Transaction): void {
+    if (
+      tx.TransactionType === 'Payment' &&
+      typeof tx.Amount !== 'string' &&
+      tx.Amount.value.includes('.') &&
+      tx.Amount.value.endsWith('0')
+    ) {
+      // eslint-disable-next-line no-param-reassign -- Required to update Transaction.Amount.value
+      tx.Amount = { ...tx.Amount }
```

Comment:
I'm pretty sure if you remove these two lines your code will work exactly the same.